lib/Ptero/Concrete/Workflow/Execution.pm

```diff
@@ -0,0 +1,109 @@
+package Ptero::Concrete::Workflow::Execution;
+
+use strict;
+use warnings FATAL => 'all';
+
+use DateTime::Format::Strptime qw();
+use DateTime qw();
+use Date::Calc "Delta_DHMS";
+use Set::Scalar;
+use Ptero::Statuses qw(is_terminal);
+
+my $DATETIME_PARSER = DateTime::Format::Strptime->new(
+    pattern => '%Y-%m-%d %H:%M:%S',
+    on_error => 'croak'
+);
+
+sub new {
+    my ($class, $hashref) = @_;
+
+    my $self = {};
+    $self->{begins} = $hashref->{begins};
```

Comment:
I wanted to leave the 'new' functions free of abstraction for the purposes of documentation.  I agree that there is a lot of boilerplate syntax here, but I think it makes it clearer what data is stored on the object at a glance.